File: src/test/scala/models/Husband.scala
package models

import javax.persistence._
import org.sporm.jpa.JPAModel

/**
 * User: YaFengLi
 * Date: 12-12-3
 * Time: 下午1:28
 */
@Entity
@Table(name = "t_husband")
case class Husband(var name: String, var age: Int) {
  @Id
  @GeneratedValue(strategy = GenerationType.TABLE, generator = "seq_t_husband")
  @TableGenerator(name = "seq_t_husband", table = "seq_t_husband", allocationSize = 1)
  var id: Long = _

  @OneToOne(fetch = FetchType.EAGER, mappedBy = "husband")
  var teacher: Teacher = _

  def this() = this(null, -1)

  def this(name: String, age: Int, teacher: Teacher) = {
    this(name, age)
    this.teacher = teacher
  }
}

object Husband extends JPAModel[Husband]

File: src/test/java/models/Husband_.java
package models;

import javax.persistence.metamodel.SingularAttribute;
import javax.persistence.metamodel.StaticMetamodel;

/**
 * User: YaFengLi
 * Date: 12-12-14
 * Time: 下午5:36
 */
@StaticMetamodel(Husband.class)
public class Husband_ {
    public static volatile SingularAttribute<Husband, Long> id;
    public static volatile SingularAttribute<Husband, String> name;
    public static volatile SingularAttribute<Husband, Integer> age;
    public static volatile SingularAttribute<Husband, Teacher> teacher;
}
